controllers/src/main/java/io/syndesis/controllers/integration/online/DeactivateHandler.java

```diff
@@ -64,11 +72,11 @@ public StatusUpdate execute(Integration integration) {
             }
         }
 
-        Integration.Status currentStatus = openShiftService.isScaled(deployment)
-            ? Integration.Status.Deactivated
-                : Integration.Status.Pending;
+        IntegrationState currentStatus = openShiftService.isScaled(deployment)
+            ? IntegrationState.Undeployed
```

Comment:
A deactivate handler should only put an integration to `deactivate`, never undeploy it. Also is must not set the current state to pending (but leave is untouched if the precondition is not matched). 

Was is currently missing here is the actual connection to the runtime to activel suspend it. This is related to the camel health check story (see syndesis board).